addons/stock/models/stock_quant.py

```diff
@@ -358,13 +361,53 @@ def _compute_complete_name(self):
         return res
 
     def _compute_current_picking_info(self):
+        """ When a package is in displayed in picking, it gets the picking id trough the context, and this function
+        populates the different fields used when we move entire packages in pickings.
+        """
+        for package in self:
+            picking_id = self.env.context.get('picking_id')
+            if picking_id:
+                package.current_picking_move_line_ids = package.move_line_ids.filtered(lambda move_line: move_line.picking_id.id == picking_id)
+                package.current_picking_id = True
+                if package.current_picking_move_line_ids:
+                    package.current_source_location_id = package.current_picking_move_line_ids[0].location_id
+                    package.current_destination_location_id = package.current_picking_move_line_ids[0].location_dest_id
+                else:
+                    package.current_source_location_id = False
+                    package.current_destination_location_id = False
+                if package.current_picking_move_line_ids.filtered(lambda ml: ml.qty_done < ml.product_uom_qty):
+                    package.is_processed = False
+                else:
+                    package.is_processed = True
+            else:
+                package.current_picking_move_line_ids = False
+                package.current_picking_id = False
+                package.is_processed = False
+                package.current_source_location_id = False
+                package.current_destination_location_id = False
+
+    def _set_current_destination_location(self):
+        """ This inverse function is there only to have the field 'current_destination_location_id' editable in
+        the view. The values set in the fields is used in 'action_toggle_processed' through context.
+        """
```

Comment:
You can set `readonly=False`. Either way I really don't agree with that, it's horrible.